project-icp/bee-pollinator-app
# Create and show April surveys via form

## Overview

This PR adds a functional April survey form that displays completed and submits new April surveys.

TODO's:
- [x] set state values from completed survey data
- [x] disabled button styling
- [x] figure out how month_year will be shared to the component (rebase on #406)
- [x] close modal on successful post or show error

Connects #379 

### Demo

<img width="935" alt="screen shot 2019-01-07 at 5 08 20 pm" src="https://user-images.githubusercontent.com/10568752/50842198-3af6f100-1334-11e9-8822-323eaac4a145.png">
<img width="930" alt="screen shot 2019-01-08 at 9 41 27 am" src="https://user-images.githubusercontent.com/10568752/50842194-3af6f100-1334-11e9-911f-6a2a95b881c3.png">

Basic error handling:
<img width="928" alt="screen shot 2019-01-07 at 5 09 57 pm" src="https://user-images.githubusercontent.com/10568752/50842200-3af6f100-1334-11e9-8c29-c81195efd2ac.png">

![completed-change](https://user-images.githubusercontent.com/10568752/50851653-66d1a100-134b-11e9-83f6-f01f6fcddbcd.gif)

### Notes

Things I'm thinking.

It would be awesome to have a custom component that all the monthly surveys inherit from, since their functionality will be the same I think. The only tricky thing is the forms are custom, and that puts a damper in reusability. --> Timebox explore in #380 

~As discussed on slack #react, I wasn't sure how to handle the requests to the survey endpoints. It doesn't make sense to me to invoke Redux since survey data is very much only used in the scope of the modal being open. This is what I have that is working, for now. Open to discussion!~ Decided on making requests from component.

## Testing Instructions

In `utils` change the `startYear` (line 192) in `getAprilSurveys` to 2018.
Visit http://localhost:8000/survey/?beekeepers and click one of the April (APRIL) surveys.
Fill out the survey if it doesn't exist, else see the results display.

Other survey listings should open an empty popup which will be subbed in with their survey form components when they exist.
